src/CppUTest/TestFailure.cpp

```diff
@@ -378,3 +385,32 @@ FeatureUnsupportedFailure::FeatureUnsupportedFailure(UtestShell* test, const cha
 
     message_ += StringFromFormat("The feature \"%s\" is not supported in this environment or with the feature set selected when building the library.", featureName.asCharString());
 }
+
+#if CPPUTEST_USE_STD_CPP_LIB
+UnexpectedExceptionFailure::UnexpectedExceptionFailure(UtestShell* test)
+: TestFailure(test, "Unexpected exception of unknown type was thrown.")
+{
+}
+
+#if defined(__GNUC__)
+static SimpleString demangle(const char* name)
+{
+    int status = -1;
+
+    std::unique_ptr<char, void(*)(void*)> demangledName {
+        abi::__cxa_demangle(name, nullptr, nullptr, &status),
+        std::free
+    };
+
+    return (status==0) ? demangledName.get() : name ;
+}
+#define DEMANGLE(name) demangle(name).asCharString()
+#else
+#define DEMANGLE(name) (name)
```

Comment:
One last thing here (I think it will be the last thing). Could you not use the macro here and just put the the demangle function always included... but then an #if in the implementation with a "return name" as alternative implementation ?

Reply:
Done